File: su18/problem102.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public IList<IList<int>> LevelOrder(TreeNode root) {
        IList<IList<int>> order = new List<IList<int>>();
        if (root == null) {
            return order;
        }
        Queue<TreeNode> queue = new Queue<TreeNode>();
        queue.Enqueue(root);
        int numCurrLevel = 1;
        int numNextLevel = 0;
        while (queue.Count() > 0) {
            IList<int> level = new List<int>();
            while (numCurrLevel > 0) {
                TreeNode node = queue.Dequeue();
                numCurrLevel--;
                level.Add(node.val);
                if (node.left != null) {
                    numNextLevel++;
                    queue.Enqueue(node.left);
                }
                if (node.right != null) {
                    numNextLevel++;
                    queue.Enqueue(node.right);
                }
            }
            order.Add(level);
            numCurrLevel = numNextLevel;
            numNextLevel = 0;
        }
        return order;
    }
}

File: su19/problem102.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def levelOrder(self, root):
        """
        :type root: TreeNode
        :rtype: List[List[int]]
        """
        order = []
        level = [root]
        while len(level) > 0:
            nextLevel = []
            levelToAdd = [elem.val for elem in level if elem is not None]
            if levelToAdd != []:
                order.append([elem.val for elem in level if elem is not None])
            for elem in level:
                if elem is None:
                    continue
                if elem.left is not None:
                    nextLevel.append(elem.left)
                if elem.right is not None:
                    nextLevel.append(elem.right)
            level = nextLevel
        return order